crates/sui-indexer/src/handlers/transaction_handler.rs

```diff
@@ -78,6 +91,9 @@ impl TransactionHandler {
             // canonical txn digest is Base64 encoded
             commit_transction_log(&mut pg_pool_conn, page.next_cursor.map(|d| d.encode()))?;
             next_cursor = page.next_cursor;
+            self.transaction_handler_metrics
```

Comment:
You have the same issue here that intervening code could exit early and skip this call.  There's two cases here, the function calls above that return results captured by `?`, and also the `await` call where if the co-routine is suspended at the await and then discarded for some reason, then the rest of its body doesn't get to run.

It's likely that the `await` case is not going to be an issue in this case, but I wanted to call it because it was a behaviour that bit us in the faucet which @666lcz spotted.